client/state/themes/test/selectors.js

```diff
@@ -0,0 +1,619 @@
+				{
+					sites: {
+						items: {
+							2916284: {
+								ID: 2916284,
+								URL: 'https://example.wordpress.com'
+							}
+						}
+					}
+				},
+				{
+					id: 'mood',
+					stylesheet: 'premium/mood'
+				},
+				2916284
+			);
+			expect( helpUrl ).to.equal( '/theme/mood/support/example.wordpress.com' );
+		} );
+
+		it( 'given a theme and Jetpack site ID, should return the correct help URL', () => {
+			const helpUrl = getThemeHelpUrl(
+				{
+					sites: {
+						items: {
+							77203074: {
+								ID: 77203074,
+								URL: 'https://example.net',
+								jetpack: true,
+								options: {
+									admin_url: 'https://example.net/wp-admin/'
+								}
+							}
+						}
+					}
+				},
+				{
+					id: 'twentysixteen',
+					stylesheet: 'pub/twentysixteen'
+				},
+				77203074
+			);
+			expect( helpUrl ).to.equal( '//wordpress.org/support/theme/twentysixteen' );
+		} );
+	} );
+
+	describe( '#getPurchaseUrl', () => {
+		it( 'given a free theme and a wpcom site ID, should return null', () => {
+			const purchaseUrl = getThemePurchaseUrl(
+				{
+					sites: {
+						items: {
+							2916284: {
+								ID: 2916284,
+								URL: 'https://example.wordpress.com'
+							}
+						}
+					}
+				},
+				{
+					id: 'twentysixteen',
+					stylesheet: 'pub/twentysixteen'
+				},
+				2916284
+			);
+			expect( purchaseUrl ).to.be.null;
+		} );
+
+		it( 'given a premium theme and a wpcom site ID, should return the purchase URL', () => {
+			const purchaseUrl = getThemePurchaseUrl(
+				{
+					sites: {
+						items: {
+							2916284: {
+								ID: 2916284,
+								URL: 'https://example.wordpress.com'
+							}
+						}
+					}
+				},
+				{
+					id: 'mood',
+					stylesheet: 'premium/mood'
+				},
+				2916284
+			);
+			expect( purchaseUrl ).to.equal( '/checkout/example.wordpress.com/theme:mood' );
+		} );
+	} );
+
+	describe( '#getThemeCustomizeUrl', () => {
+		it( 'given no theme and no site ID, should return the correct customize URL', () => {
+			const customizeUrl = getThemeCustomizeUrl( {} );
+			expect( customizeUrl ).to.equal( '/customize/' );
+		} );
+
+		it( 'given a theme and no site ID, should return the correct customize URL', () => {
+			const customizeUrl = getThemeCustomizeUrl(
+				{},
+				{
+					id: 'twentysixteen',
+					stylesheet: 'pub/twentysixteen'
+				}
+			);
+			expect( customizeUrl ).to.equal( '/customize/' );
+		} );
+
+		it( 'given a theme and wpcom site ID, should return the correct customize URL', () => {
+			const customizeUrl = getThemeCustomizeUrl(
+				{
+					sites: {
+						items: {
+							2916284: {
+								ID: 2916284,
+								URL: 'https://example.wordpress.com'
+							}
+						}
+					}
+				},
+				{
+					id: 'twentysixteen',
+					stylesheet: 'pub/twentysixteen'
+				},
+				2916284
+			);
+			expect( customizeUrl ).to.equal( '/customize/example.wordpress.com?theme=pub/twentysixteen' );
+		} );
+
+		// FIXME: In implementation, get rid of `window` dependency.
```

Comment:
I think ideally we'd have the router store the current URL somewhere in `state/ui` and read it from there. (That could/should be done isomorphically, too.)